Adds "armoury", a command which interacts with the World of Warcraft Armoury API

Zarthus opened the pull request:
I'm planning to extend on this later on, but for now, this command works as is. It's very basic but does what it needs to do, as the API (and game) is rather big, and has a lot you can retrieve data from.

For now, all we fetch is: `Name`, `Level`, `Gender`, `Class`, `Realm`, `Achievement Points`, `Honourable Kills`.

Syntax: `.armoury [realm] [character name] [region = eu]`,  also supports the American spelling `.armory`
`Realm`: Any realm from any (supported) region. You'll have to specify the region if it is not Europe. Use dashes or underscores if your realm contains a space.  
`Character name`: Name of your character, it can be any unicode name, but may not be longer than 18 characters, or contain digits.  
`Region`: By default Europe, supported regions are `eu (Europe)`, `us (United States)`, `sea (South East Asia)`, `kr (Korea)`, `tw (Taiwan)`, which should cover all of them.

An example message string by `.armoury` is:

```
(19:03:40) <Zarthus> .armoury stormrage zarthus  
(19:03:41) <cloudthus> (Zarthus) Zarthus is a level 90 female Death Knight on Stormrage with 5515 achievement points and 124 honourable kills. Armoury Profile: http://is.gd/opjtNZ

(19:48:14) <@Zarthus> .armoury borean_tundra shuliay us
(19:48:15) <cloudthus> (Zarthus) Shuliay is a level 90 female Shaman on Borean Tundra with 9410 achievement points and 4646 honourable kills. Armoury Profile: http://is.gd/kGbYmh

(10:00:01) <@Zarthus> .armoury hueheuh xx
(10:00:01) <cloudthus> (Zarthus) Realm not found.

(10:00:08) <@Zarthus> .armoury stormrage xxxxtuhs
(10:00:08) <cloudthus> (Zarthus) Character not found.
```

[Visual Representation](http://i.imgur.com/fKm0eZU.png), [Second Visual Representation](http://i.imgur.com/OjoRUAB.png)

It's been tested, but I'm still very much new to Python, so there may be certain optimizations that I skipped over. I'm open to critique.

Commit:
Add plugin wow.py, which uses the World of Warcraft armoury API to fetch data, commands: .armory, .armoury

Commit:
wow.py: Finishing touches, improved regex, clearer error messages, few bugfixes.

Commit:
Identation: tabs to spaces, add Zarthus to CONTRIBUTORS.

Commit:
Finishing touches: Remove trailing whitespace.

literalplus commented:
:+1 for the branch name

-- Above message was sent from my mobile phone. There may and will be all
kinds of errors included. For free!

Commit:
Make use of the 'requests' instead of 'http' library.

We now use the requests library, which doesn't throw exceptions on 404 unless you tell it to. Helping give more detailed error messages and resolving one small issue the code had.
